llms/llm.go

```diff
@@ -229,6 +229,11 @@ func (l *LLM) turn(ctx context.Context, updateChan chan<- Update) (bool, error)
 			l.TotalUsage.Add(usage)
 			l.TrackUsage(ctx, usage, success)
 		}()
+	} else {
+		defer func() {
+			usage := stream.Usage()
+			l.TotalUsage.Add(usage)
+		}()
```

Comment:
Instead of two `defer` blocks, let's just assign `trackUsage := l.TrackUsage` and then check `if trackUsage != nil` inside of the func.